### DISCUSSION TITLE
Flutter: Wallet connect v2: Metamask signature request doesn't appear on metamask app

### DISCUSSION THREAD

**Discussion by zo-pinkal** [WalletConnect/walletconnect-monorepo#3324] [Developer Support]
### Description:
I have used this demo [[flutter_web3_demo](https://gitlab.com/graflr/flutter_web3_demo)] as a reference to integrate Walletconnect v2 in my app. I am able to open connect dialog in Metamask app for connection and able to get a successful connection but when I used to call signature requests it is not able to open in the Metamask app. Please check the below details which I used currently and below code sample which I used to request a signature.

### **Details:**
- **walletconnect_flutter_v2:** ^2.0.14
- **Metamask Android app version:** 7.3.0
- **Android OS:** 13

### Code:
**Initialize:**
```dart
PairingMetadata pairingMetadata = const PairingMetadata(
    url: kAppUrl,
    name: kAppName,
    description: kAppDescription,
    icons: kAppIcons,
  );

_web3App = await Web3App.createInstance(
        projectId: walletConnectProjectId,
        metadata: pairingMetadata, 
);

_signClient = await SignClient.createInstance(
        projectId: walletConnectProjectId,
        metadata: pairingMetadata, 
);
```
**Connect wallet:**
```dart
 resp = await _signClient.connect(requiredNamespaces: {
      'eip155': eip155RequiredNamespace,
    });
```

**Request for signature:**
```dart
    final AuthRequestResponse authReq = await _web3App.requestAuth(
      params: AuthRequestParams(
        aud: AppEnvironment.defaultWalletURL,
        domain: AppEnvironment.defaultWalletDomain,
        statement: 'Authenticate with App',
        nonce: nonce,
        chainId: 'eip155:1',
        nbf: notBefore.toIso8601String(),
        exp: expires.toIso8601String(),
        resources: ['https://eips.ethereum.org/EIPS/eip-4361'],
      ),
    );
```

**Comment by mshahjahan-mufin** [comment]
Please check, you are using Web3App but connecting on the base of SignClient and then calling Web3App for auth. You have to either use Web3App or SignClient, not both.
